### PR TITLE
docs: Add howto guide on how to read/write to filesystem.

### PR DESCRIPTION
**Pull request by qchong**
@collinjackson ptal

Fixes https://github.com/flutter/flutter/issues/5392
Staged preview: https://flutter-io-staging-69a8a.firebaseapp.com/saving-files/

### CODE REVIEW

**Comment by sethladd** [comment]
This shows more than saving. Maybe a title like Reading and Writing Files ? Also, reading-writing-files for moar SEO :)

**Comment by collinjackson** [comment]
fix analyzer warnings by replacing void with Future<Null>.

lgtm otherwise

File: _data/sidebars/home_sidebar.yml [hunk 1]
```diff
@@ -86,6 +86,10 @@ entries:
       url: /platform-services/
       output: web
 
+    - title: Saving Files
+      url: /saving-files/
```
Comment (sethladd) [inline _data/sidebars/home_sidebar.yml]: Should we change this to reading-writing-files ?

File: index.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ Learn how to accomplish specific development tasks with Flutter.
  - [Debugging Flutter apps](debugging)
  - [Upgrading Flutter](upgrading)
  - [Accessing Platform and Third-Party Services](platform-services)
+ - [Saving Files](saving-files)
```
Comment (sethladd) [inline index.md]: Should we update this to [Reading and writing files](reading-writing-files) ?

**Pull request merged**